### PR TITLE
add /sandbox/public_token/create

### PR DESCRIPTION
**Pull request by michaelckelly**
- Add `/sandbox/public_token/create` endpoint

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -532,6 +536,13 @@ declare module 'plaid' {
     getCategories(cb: Callback<CategoriesResponse>): void;
 
     resetLogin: AccessTokenFn<ResetLoginResponse>;
+
+    // sandboxCreatePublicToken(String, Array<sring>,Object?, Function)
+    sandboxPublicTokenCreate(institutionId: string,
+                    initialProducts: Array<string>,
+                    options: Object,
+                    cb: Callback<SandboxPublicTokenCreateResponse>
+    ): Promise<SandboxPublicTokenCreateResponse>;
```
Comment (ntindall) [inline index.d.ts]: we need to define two flavors of this function, one for the CPS style, and one for the promisified variant

```
// sandboxCreatePublicToken(String, Array<sring>,Object?, Function)
sandboxPublicTokenCreate(
  institutionId: string,
  initialProducts: Array<string>,
  options: Object,
  cb: Callback<SandboxPublicTokenCreateResponse>,
): void;

// sandboxCreatePublicToken(String, Array<sring>,Object?, Function)
sandboxPublicTokenCreate(
  institutionId: string,
  initialProducts: Array<string>,
  options: Object,
): Promise<SandboxPublicTokenCreateResponse>;
```

when they are both defined, typescript will pick the one that compiles.

File: index.d.ts [hunk 2]
```diff
@@ -532,6 +536,13 @@ declare module 'plaid' {
     getCategories(cb: Callback<CategoriesResponse>): void;
 
     resetLogin: AccessTokenFn<ResetLoginResponse>;
+
+    // sandboxCreatePublicToken(String, Array<sring>,Object?, Function)
+    sandboxPublicTokenCreate(institutionId: string,
+                    initialProducts: Array<string>,
+                    options: Object,
```
Comment (ntindall) [inline index.d.ts]: nit: should we provide a typing for the `options`? we seem to be somewhat inconsistent about this... so maybe its "out of scope".

File: lib/PlaidClient.js [hunk 3]
```diff
@@ -332,4 +332,16 @@ Client.prototype.getCategories =
...
+// sandboxPublicTokenCreate(String, Array, Object?, Function) - sandbox only
+Client.prototype.sandboxPublicTokenCreate =
+function(institution_id, initial_products, options, cb) {
+  return this._authenticatedRequest({
+    path: '/sandbox/public_token/create',
+    body: {
+      institution_id: institution_id,
+      initial_products: initial_products,
+    },
+  }, options, cb, true);
```
Comment (ntindall) [inline lib/PlaidClient.js]: are there actually`options` for this route?

I'm realizing the fact that `_authenticatedRequest` [injects options onto the request](https://github.com/plaid/plaid-node/blob/master/lib/PlaidClient.js#L48) in this way is somewhat confusing...

pretty sure I devised this scheme to support the variadic arguments behavior when you use the callback style... I wonder if anyone is benefitting from that...

**Comment by ntindall** [comment]
looks good after we fix the typings @michaelckelly !

File: index.d.ts (hunk 1 above)
Comment (michaelckelly) [reply index.d.ts]: Thanks @ntindall - should be ⚡️'d!

File: lib/PlaidClient.js (hunk 3 above)
Comment (michaelckelly) [reply lib/PlaidClient.js]: 👍

**Pull request merged**